### ISSUE TITLE
WIP Task Tactical Marker crashes DCS 1.5.7.9459

### ISSUE THREAD

**Issue by Ramsay58** [FlightControl-Master/MOOSE#698]
Testing the new 'Mark Task on Map' feature found in moose.lua build timestamp: 20170914_1227 caused Steam DCS 1.5.7.9459 to crash to desktop.

Doing the same test in DCS 1.5.6.5199 did not cause DCS to crash to desktop but did not create a 'Tactical Marker' either.

Given that the feature did not crash 1.5.6, I suspect a problem with 1.5.7.9459 rather than MOOSE caused the CTD.

I could only find 'Mark Task on Map' under the 'Join Task' sub menu.
![moose_jointask_tactical marker_screen_170915_021758](https://user-images.githubusercontent.com/31033413/30465067-e81e60da-99cc-11e7-830e-492579041269.jpg)

Opening the Status menu worked but was still using METRIC despite having set user preference to IMPERIAL.
![moose_jointask_report_still_metric_despite_imperial_settings_screen_170915_022438](https://user-images.githubusercontent.com/31033413/30465145-3a1b5758-99cd-11e7-818d-1712581171ab.jpg)

Selecting 'Mark Task on Map' crashed DCS 1.5.7.9459 or closed the menu but did not place a marker in 1.5.6
![moose_jointask_marktask_crashes_dcs_screen_170915_021758](https://user-images.githubusercontent.com/31033413/30465204-9eac65e0-99cd-11e7-8c64-df9af535d95e.jpg)

If instead, I joined the task
![moose_jointask_works_screen_170915_022531](https://user-images.githubusercontent.com/31033413/30465224-bf019c48-99cd-11e7-914c-3553cd0c4b2f.jpg)

It removed the option to place a marker and there didn't seem to be a duplicate elsewhere
![moose_jointask_removes_option_to_mark_map_screen_170915_022850](https://user-images.githubusercontent.com/31033413/30465311-4efaba1e-99ce-11e7-8a20-471d3442a2a4.jpg)

IIRC in DCS 1.5.6 other planned tasks still had the option to join/mark them, so it's possible the option was still in 1.5.7 for other tasks but I was looking in the wrong place.

Tested using
Steam DCS 1.5.7.9459 and a backup of Steam DCS 1.5.6.5199
MOOSE build timestamp: 20170914_1227
Steam M-2000C hot start at Krasnodar (PASHKOVSKY)
OS Win 10 pro x64
Mods TacView 1.6.4 beta1

Sample 1.5.7 dcs.log after selecting 'Mark Task on Map'
[dcs_log.zip](https://github.com/FlightControl-Master/MOOSE/files/1305221/dcs_log.zip)

**Comment by FlightControl-Master** [comment]
@Ramsay58 The reason is that target descriptions to be put into the marking box may become a bit too long, and will crash DCS. I've had many crashes like this already and still am searching for the root cause. It seems it has to do with the text not fitting into the boundary boxes of the marking. I will try to make the target descriptions a bit shorter and will cap the text to about 255 characters, which may be one of the causes.

**Comment by FlightControl-Master** [comment]
> @Ramsay58 Testing the new 'Mark Task on Map' feature found in moose.lua build timestamp: 20170914_1227 caused Steam DCS 1.5.7.9459 to crash to desktop.

Fixed the marking:

![screen_170915_140210](https://user-images.githubusercontent.com/13690775/30487732-9c0e7276-9a34-11e7-826f-6026c0d0d32a.jpg)

I have made the text shorter. I have removed all the details of the targets, it is not needed because those are shown on the reports. But it shows speed and direction information. I need to optimize a bit the text in case speed is 0. I should mention stationary instead of at 0km/h. And the degrees need to be with 0 decimals. I also need to work out a better algorithm that if 
- some units of the group are moving, 
- and not moving in the same direction (that is the degrees of movement is more or less the same) 
then the heading cannot be determined.

**Comment by FlightControl-Master** [comment]
> @Ramsay58 Opening the Status menu worked but was still using METRIC despite having set user preference to IMPERIAL.

I've been updating now the logic and fixing this forever!

![screen_170915_170002](https://user-images.githubusercontent.com/13690775/30489495-9c4ed1ce-9a37-11e7-8d86-40c4c78a6902.jpg)

The coordinates WILL be displayed in BR if you have selected your A2G settings to work in BR format.

FC

**Comment by FlightControl-Master** [comment]
Download the latest moose.lua as usual from the release page:

https://github.com/FlightControl-Master/MOOSE/releases

**Comment by Ramsay58** [comment]
BRA reported in IMPERIAL with IMPERIAL user settings.
![assigned_taskreport_with_imperial_screen_170916_001103](https://user-images.githubusercontent.com/31033413/30507778-18bdb9de-9a81-11e7-8c5f-5b6b2c066fbe.jpg)

And Join Task Tactical Markers worked in DCS 1.5.7.9459
![moose_new_jointask_marktask_screen_170916_000928](https://user-images.githubusercontent.com/31033413/30507780-1d1233c0-9a81-11e7-9a6f-cb51f4ed02cc.jpg)

looks good.

**Test Mission debrief**
I switched the A2G user preference to LL DDM to enter INS target coordinates from the Detailed Report and the target MSL from the F10 map, before takeoff and switched to BRA for the flight. LL DDM agreed with the placed Task Marker as it should.

As the mission progressed, the Task Marker was able to show how the M-2000C INS coordinates drifted and the effect of an INS update.

I later switched to another task in the same area but placed the Task Marker to read LL DDM directly from the F10 map as it was easier (while flying) than switching user A2G preference, calling up the detailed task report for LL DDM for the INS and then switching A2G back to BRA.

I think this issue can be closed unless there are any other reports.

Tested using MOOSE build timestamp: 20170915_1940
